### PR TITLE
prometheus push metrics

### PR DESCRIPTION
**Pull request by shenli**
@coocood @siddontang  PTAL
Refer: https://github.com/pingcap/pd/pull/325

### CODE REVIEW

**Comment by siddontang** [comment]
LGTM

File: tidb-server/main.go [hunk 1]
```diff
@@ -32,21 +32,25 @@ import (
...
+	port            = flag.String("P", "4000", "tidb server port")
+	statusPort      = flag.String("status", "10080", "tidb server status port")
+	lease           = flag.Int("lease", 1, "schema lease seconds, very dangerous to change only if you know what you do")
+	socket          = flag.String("socket", "", "The socket file to use for connection.")
+	enablePS        = flag.Bool("perfschema", false, "If enable performance schema.")
+	reportStatus    = flag.Bool("report-status", true, "If enable status report HTTP service.")
+	logFile         = flag.String("log-file", "", "log file path")
+	joinCon         = flag.Int("join-concurrency", 5, "the number of goroutines that participate joining.")
+	metricsAddr     = flag.String("metrics-addr", "", "prometheus pushgateway address, leaves it empty will disable prometheus.")
+	metricsInterval = flag.String("metrics-interval", "0s", "prometheus client push interval, set \"0s\" to disable prometheus.")
```
Comment (zimulala) [inline tidb-server/main.go]: Use `metricsInterval` and `metricsInterval`.

Comment (siddontang) [inline tidb-server/main.go]: not disable prometheus, but push.

**Comment by shenli** [comment]
@zimulala PTAL

**Comment by zimulala** [comment]
LGTM

File: tidb-server/main.go [hunk 2]
```diff
@@ -131,3 +135,33 @@ func main() {
...
+
+// PushMetric pushs metircs in background.
+func pushMetric(addr string, interval time.Duration) {
+	if interval == zeroDuration || len(addr) == 0 {
+		log.Info("disable Prometheus push client")
+		return
+	}
+	log.Info("start Prometheus push client")
+	// TODO: TiDB do not have uniq name, so we use host+port to compose a name.
+	name := fmt.Sprintf("TiDB-%s:%s", *host, *port)
```
Comment (overvenus) [inline tidb-server/main.go]: TiKV uses `TiKV_1`, I think it better be to `"TiDB_%s:%s"`.

**Comment by shenli** [comment]
@overvenus PTAL

**Comment by overvenus** [comment]
LGTM

**Comment by coocood** [comment]
LGTM

**Pull request merged**